### PR TITLE
[ICDS] Changed place of allert in the ICDS Dashboard

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @calellowitz @Rohit25negi 

I changed the place of HQ alerts in the ICDS dashboard, also I had to change an HTML tree to fix a problem with ```position: fixed``` in top menus. I attached screenshot how the dashboard looks after my change.

Need QA: yes
Environment: ICDS
Locally tested: yes

Regards,
Lukasz

icds-alert

### CODE REVIEW

File: custom/icds_reports/static/js/angular-services/baseControllers.service.js [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+        BaseController: function ($scope, $routeParams, $location, locationsService, userLocationId,
             storageService, haveAccessToAllLocations, haveAccessToFeatures) {
             var vm = this;
             if (Object.keys($location.search()).length === 0) {
                 $location.search(storageService.getKey('search'));
             } else {
                 storageService.setKey('search', $location.search());
             }
+
+            vm.showInfoMessage = function () {
```
Comment (orangejenny) [inline custom/icds_reports/static/js/angular-services/baseControllers.service.js]: Could this function be extracted into a separate file and shared among the directives?

File: custom/icds_reports/templates/icds_reports/icds_app/awc-opened-yesterday.directive.html [hunk 2]
```diff
@@ -19,6 +19,11 @@
             </div>
         </div>
     </div>
+    <div class="alert-container" ng-if="$ctrl.showInfoMessage()">
```
Comment (orangejenny) [inline custom/icds_reports/templates/icds_reports/icds_app/awc-opened-yesterday.directive.html]: Could this block be extracted into a partial template?

**Comment by lwyszomi** [comment]
Hi @orangejenny 

I created a new directive for info-message.

Please let me know if you have any questions.

Regards,
Lukasz

**Review by orangejenny — APPROVED** [review]
Thanks Lukasz, this is much more concise.

**Comment by orangejenny** [comment]
@calellowitz @Rohit25negi I don't know enough about dashboard process to know whether or not to merge this, since it says it needs QA. Leaving that to you to decide/merge.

**Comment by Rohit25negi** [comment]
@lwyszomi  Did you try it on small resolution screen where person has to scroll up-down. Earlier that was messing for me, when I tried putting banner. If not can you test this?

File: custom/icds_reports/static/js/directives/info-message/info-message.directive.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        var currentMonth = new Date().getMonth() + 1;
+        var currentYear = new Date().getFullYear();
+        if (!$location.path().startsWith("/fact_sheets") && !$location.path().startsWith("/download") &&
+            selectedMonth === currentMonth && selectedYear === currentYear &&
+            (new Date().getDate() === 1 || new Date().getDate() === 2)) {
+            vm.lastDayOfPreviousMonth = moment().set('date', 1).subtract(1, 'days').format('Do MMMM, YYYY');
+            vm.currentMonth = moment().format("MMMM");
+            return true;
+        }
+        return true;
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: it looks returning returns needs to be corrected. This function is always returning true.

Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/info-message/info-message.directive.js]: This will lead the banner to always show

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Left one small change request and one test request. Else Looks good.

**Comment by lwyszomi** [comment]
Hi @Rohit25negi,

Thanks you for reminding me that this should looks good on the small resolution, I forgot about this.

small resolution:
small-resolution

big resolution:
big-resolution

Please let me know if you have any questions.

Regards,
Lukasz

**Comment by lwyszomi** [comment]
hold of merge

**Comment by Rohit25negi** [comment]
adding do not merged tag based on lukasz last comment

**Comment by lwyszomi** [comment]
@Rohit25negi ready for review and merge.

**Comment by Rohit25negi** [comment]
Will there be two alerts on Icds if its 1st of the month and because of some error on dashboard we also applied one  alert message?

**Review by Rohit25negi — APPROVED** [review]
Looks good to me

**Comment by Rohit25negi** [comment]
This looks good. Only one questions. Is this needs to be under FF or to be released without FF?

**Comment by Rohit25negi** [comment]
Ok Merging this based on the go ahead without FF on the ticket, https://dimagi-dev.atlassian.net/browse/ICDS-463.

**Comment by Rohit25negi** [comment]
Also @lwyszomi  its always good to add the ticket in the PR to add more context who don't know. I would really appreciate If you can do that in future PRs.

**Pull request merged**